sources/api/migration/migration-helpers/src/common_migrations.rs

```diff
@@ -78,7 +79,7 @@ impl Migration for AddPrefixMigration {
         let settings = input
             .data
             .keys()
-            .filter(|k| k.starts_with(self.0))
+            .filter(|k| self.0.iter().any(|prefix| k.starts_with(prefix)))
```

Comment:
nit: This might be a weird edge case... but consider this:

Suppose we have an existing setting prefix called `settings.doggo` that we want to keep, and we want to add a migration for adding `settings.dog`. The resulting backwards migration would remove both `settings.doggo` and `settings.dog`.

Can we filter the prefix only if it ends with a '.'?

Reply:
I think that's a case where we'd want to be extra careful and consider writing our own migration rather than using a helper.  But to filter only if something ends with a `.`, you can just include the `.` when you're creating the migration, e.g. `AddPrefixesMigration(vec!["settings.kernel.sysctl."])`, but I thought it was clearer not to in this case.